test/unit/test_configmanager.py

```diff
@@ -96,13 +103,175 @@ def test_simple_config_read(tmp_files):
         name="connections",
         parse_str=parse,
     )
+    settings_parser = ConfigManager(
+        name="settings",
+    )
+    settings_parser.add_option(
+        name="output_format",
+        choices=("json", "yaml", "toml"),
+    )
+    TEST_PARSER.add_subparser(settings_parser)
+    assert TEST_PARSER["connections"] == {
+        "snowflake": {
+            "account": "snowflake",
+            "user": "snowball",
+            "password": "password",
+        }
+    }
+    assert TEST_PARSER["settings"]["output_format"] == "yaml"
+
+
+def test_simple_config_read_sliced(tmp_files):
+    """Same test_simple_config_read, but rerads part of the config from another file."""
+    tmp_folder = tmp_files(
+        {
+            "config.toml": dedent(
+                """\
+                [settings]
+                output_format = "json"
+                """
+            ),
+            "connections.toml": dedent(
+                """\
+                [snowflake]
+                account = "snowflake"
+                user = "snowball"
+                password = "password"
+                """
+            ),
+        }
+    )
+    TEST_PARSER = ConfigManager(
+        name="root_parser",
+        file_path=(
```

Comment:
I wonder if it makes sense for entries in this tuple to be [nametuples](https://docs.python.org/3/library/collections.html#collections.namedtuple).  It might make element accessing in the code above more readable.